### DISCUSSION TITLE
Adding multiple colums of shortcuts.

### DISCUSSION THREAD

**Discussion by Piwielle** [Hofknecht/SystemTrayMenu#379] [Ideas]
Hey. So, I moved to Windows 11, and I dislike the new start menu.
Alternatives exist, like Start11, but I also dislike them, mainly because they're painfully slow.

I've been using STM, and it's been serving me well. Congratulations.

However, I would personally love a way to add multiple columns of shortcuts, to be able to sort shortcuts in categories, without having to scroll. That would make it perfect for me anyway. Here's a crude paint of how that might look with 2 colums.

![image](https://user-images.githubusercontent.com/47955122/167270937-2bc7cb33-71d7-451d-8228-5e01136fa82d.png)

I'm note quite sure how this could work with the current way things are done. Maybe in the Folders settings, there would be a new checkmark "Add this folder as another column", which would then spawn a new column next to the existing one(s) ? This seem like a good way of adding the feature, but there's probably a better way to do it.

Also not quite sur on how it would work with browsing folders. Should it open on top of others columns, or on the right of every column ? The second option might be a pain while using a mouse if you mosue over another folder while going to the subfolder. I don't know what the best option would be.

Anyway, thanks for the work you've done so far, and hopefully someone likes this idea too :)
Have a nice day.

**Comment by Hofknecht** [comment]
@Piwielle 
Thank you for the suggestion!
It's a very interesting way. It would also be a workaround for #374 or #252 because you have several menus that work similarly to one.

The implementation would not be that easy or we are considering whether there should be an adjustable number of columns.
I can't quite imagine how to best configure this in the settings menu

Behavior also needs to be carefully considered.
In the second search you can then switch with tab, but you can also switch with tab to a possibly open sub-menu. Maybe it should be one search for both menus. Below there is the files counter, this should perhaps still only be for the one main menu.
We also have to think again about the question of where exactly sub-menus should appear. A sub-menu then hides the second column.

It could be that it all doesn't seem so native anymore.

Right now I think we could also have a kind of storage as a second column, which is organized similar to these cell phone apps and thus represents an additional quick-click menu

![2022-05-14 15_28_05-Adding multiple colums of shortcuts  · Discussion #379 · Hofknecht_SystemTrayMen](https://user-images.githubusercontent.com/52528841/168427858-c793c3f9-3467-4bca-9963-98b141aab1d7.png)

We need to think about this and whether we can imagine it and what the best nativ way is.

**Comment by Piwielle** [comment]
Ah, I see what you mean regarding the settings implementation. More than one extra column could be an issue. It can be solved, but then it adds a lot of considerations.
And behavior is also complicated, and like you said, it might end up being a lot of work for something that would end up not native.

 A second colum which could effectively be used as a shelf for applications would also honestly be great. That's pretty much how Windows 10 does it, and in my personal opinion, it's a good design.
Thinking about it, the important features to me are : 

• Being able to add as many icons as I'd like on a menu.
• Being able to sort them in groups (ideally).

The grouping might not be very easy to implement ? Not sure. I could barely write a hello word to save my life, but dynamic sizes and such do not sound easy.
Fixed group sizes with settings sounds easier to make, but then I guess settings will be a mess (group dimensions, names, number of groups, and so on).

Maybe the groups should not be a feature, and it should be everything you add, like the app drawer of some phones yeah.

Here's a mockup of how a shelf with groups and such might look like (one can dream), and where in settings you could add the enable toggle. This assumes that it magically gets dynamic sizes and names, one can dream :D
![kZgd0C5](https://user-images.githubusercontent.com/47955122/168447746-93d0f198-f052-41a8-90e2-338b42497731.png)

**Comment by Hofknecht** [comment]
That in itself is very time-consuming, but when I have time I'll come back to the topic.
I'm marking it as a future implementation.
